### DISCUSSION TITLE
Admin rights are required

### DISCUSSION THREAD

**Discussion by AnzorSh** [wqweto/VbAsyncSocket#28] [Ideas]
Maybe you know about it. The connection fails if the server is running with restricted user rights.

> server
> ctxServer_ConnectionRequest, requestID=924, RemoteHostIP=127.0.0.1, RemotePort=58751, Time 7/7/2022 4:22:41 AM
> ctxServer_Error:  -2146500593	Object already exists., Time 7/7/2022 4:22:41 AM
> ctxServer_CloseEvent, Connections=0, Time 7/7/2022 4:22:41 AM

> client
> ctxWinsock_Error: -2146499839	The handle specified is invalid, Time 7/7/2022 4:30:27 AM

Will this change in the future?

**Comment by wqweto** [comment]
I just compiled `Project1.vbp` in `test/Winsock` and started `Project1.exe` under non-admin user and clicking `HTTPS Server` button works as expected. Additionally tested listening on port 888 (under 1024) and on IP 0.0.0.0 (not 127.x.x.x from loopback adapter) still works for non-admins as expected.

This might be a problem outside this repo's code but ping me if you find out why non-admin users are failing.

Btw, if you are serious about using this for anything TLS you should consider migrating your server-side code to `cTlsSocket` class instead of using the Winsock replacement clutch. Using a form and a control-array of user-controls brings nothing for a server application and is better to not perpetuate as a design pattern.

**Comment by AnzorSh** [comment]
got it, mdTlsThunks.bas works under regular user, not Guest
**mdTlsNative.bas** work only with admin rights

While this is just a hobby, GUI utility, by no means invisible.
In the future, I plan to do able to edit the script of the operation.
[https://youtube.com/shorts/8sCtqfPZpIs?feature=share](YouTube)
It is unlikely that I will be able to earn money by developing on VB6, no matter what cute interfaces I draw))

**Reply by wqweto** [reply]
10x for the info. I'll check which API is failing and if there is a way to mitigate failures.

**Comment by wqweto** [comment]
I did some cleanup in latest commits around permissions problems w/ `CryptAcquireContext` API and as a result thunks based backend now works under both guest and non-admin accounts. 

Unfortunately it seems Schannel is unable to access private keys for server-side and client certificates under temporary profiles so server-side native backend is impossible under guests accounts and the same problem exists when using client certificates.

Native backend works successfully under non-admin users in both server-side and client certificates variants now though.
